### PR TITLE
Add the possibility to set default feature gates

### PR DESCRIPTION
**Pull request by jean-edouard**
**What this PR does / why we need it**:
Since feature gates are defined in the KubeVirt CR, they default to an empty list and that can't be changed.
This PR adds the possibility to define "feature_gates" as a coma-separated list in the provider config.
The feature gates will be reflected accordingly in the kubevirt-cr yaml file.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
This is a best effort PR, which is far from perfect, any improvement is more than welcome.
The problem is that we expect the yaml templates to be valid yaml, which is usually not the case.
It works when the template is simple and all values are strings, which seemed to be the case until now.

Important: If this PR is rejected, the Makefile hunk and the resource-generator usage fix should still be considered.

**Release note**:

```release-note
Default feature gates can now be defined in the provider configuration.
```

### CODE REVIEW

**Comment by acardace** [comment]
/lgtm

How you treat the range template along the way is a bit convoluted but the patch certainly makes the best of it, thank you!
+1 The comment to explain how you treat the template in `manifest-templator.go` is very useful.

**Comment by jean-edouard** [comment]
@acardace those convoluted lines bothered me enough to give this another shot (force-pushed).
This time, the whole logic is in 1 place (`tools/util/marshaller.go`), and the comment has been improved to better explain what's happening and why.
Also if in the future we want to handle more lists, the marshaller code can now be reused pretty much as-is.

**Comment by acardace** [comment]
> @acardace those convoluted lines bothered me enough to give this another shot (force-pushed).
> This time, the whole logic is in 1 place (`tools/util/marshaller.go`), and the comment has been improved to better explain what's happening and why.
> Also if in the future we want to handle more lists, the marshaller code can now be reused pretty much as-is.

Definitely looks better now, thanks!

File: tools/resource-generator/resource-generator.go [hunk 1]
```diff
@@ -32,9 +33,10 @@ import (
 )
 
 func main() {
-	resourceType := flag.String("type", "", "Type of resource to generate. vmi | vmipreset | vmirs | vm | vmim | kv | rbac | priorityclass")
```
[the remark this answers is not in the document]
Comment (stu-gott) [reply tools/resource-generator/resource-generator.go:35]: Sorry, I just noticed this comment was still pending!

[the remark this answers is not in the document]
Comment (jean-edouard) [reply tools/resource-generator/resource-generator.go:35]: I guess so yeah! If called with any other value than `kv | kv-cr | operator-rbac | priorityclass`, the program will `panic(fmt.Errorf("unknown resource type %s", *resourceType))`. I did not go back in git history to see what happened, but I assume these were valid options at some point!

**Review by stu-gott — COMMENTED** [review]
/approve

**Pull request merged**